src/actions/attack.cpp

```diff
@@ -183,7 +183,7 @@ battle_context_unit_stats::battle_context_unit_stats(const unit& u,
 			resources::gameboard->units(), resources::gameboard->map(), u_loc, u.alignment(), u.is_fearless());
 
 	// Leadership bonus.
-	int leader_bonus = under_leadership(units, u_loc).first;
+	int leader_bonus = under_leadership(units, u_loc, weapon).first;
```

Comment:
It might be nice if you could pass in `opp_weapon` too for a `[filter_opponent_weapon]` tag.